packages/cookies/src/response-cookies.ts

```diff
@@ -14,8 +14,11 @@ export class ResponseCookies {
 
   constructor(responseHeaders: Headers) {
     this._headers = responseHeaders
-    // @ts-expect-error See https://github.com/whatwg/fetch/issues/973
-    const headers = this._headers.getAll('set-cookie')
+    const headers =
+      // @ts-expect-error See https://github.com/whatwg/fetch/issues/973
+      this._headers.getAll?.('set-cookie') ??
+      this._headers.get('set-cookie')?.split(', ') ??
```

Comment:
This might not be sufficient as pointed out in https://vercel.slack.com/archives/C03ENM5HB4K/p1668180220270349?thread_ts=1668173326.539069&cid=C03ENM5HB4K will investigate